### DISCUSSION TITLE
Getting The Selected Item from a Combo

### DISCUSSION THREAD

**Discussion by Godlike-Productions** [RomanKubiak/ctrlr#203] [Show and tell]
Strangely I couldn't find this one over at the old forums.  Lots of stuff on setting them, but nothing on getting the value.

```
panel:getCombo("upperProgramCombo"):getSelectedId()
```

**Comment by Ted-juh** [comment]
Does this work?

comboId = panel:getcombo(name):getSelectedId()

If comboId == 0 then
-- do something
end

Or you have to use getSelectedItemIndex, can't remember.

**Reply by Godlike-Productions** [reply]
That will work.

This is my entire method.

I'm not sure if there is an index method as well, because Id is essentially Index.

```
updatePatchName = function(--[[ CtrlrModulator --]] mod, --[[ number --]] value, --[[ number --]] source)

    local patchSelected = panel:getCombo("upperProgramCombo"):getSelectedId()
    patchNameLabel:getComponent():setPropertyString ("uiLabelText", jupiterBank[patchSelected].name)

end
```

**Reply by dnaldoog** [reply]
But these are they!

**Comment by Ted-juh** [comment]
int getSelectedId();
	int getSelectedItemIndex();
	void setSelectedId(const int id, const bool dontNotify);
	void setSelectedItemIndex(const int index, const bool dontNotify);

From the comboBox.h file.

**Reply by dnaldoog** [reply]
Here are a few functions I use:

<pre><code>
function returnComboSelectedItemIndex(myCombo)
	-- Your method code here
    local c = panel:getComboComponent(myCombo)
        local combo = c:getOwnedComboBox()
return combo:getSelectedItemIndex()
end

function returnComboElementCount(myCombo)
	-- Your method code here
return panel:getModulatorByName(myCombo):getProperty("modulatorMax")
end

function returnSelectedPatchNameFromCombo(myCombo)
	-- Your method code here
    local c = panel:getComboComponent(myCombo)
        local combo = c:getOwnedComboBox()
            return combo:getText() 
end

</code></pre>

**Reply by Godlike-Productions** [reply]
Also Atom gives some good examples of multi level combos in the DEMO - popup menu panel, which comes with ctrlr

**Comment by dobo365** [comment]
As usual, the full list of capabilities can be found in the Juce API.
Look at: https://docs.juce.com/master/classComboBox.html
But this is also a full model and you need to check what has been ported to Ctrlr by testing the different methods/functions.

**Reply by Godlike-Productions** [reply]
I still get confused with the c++ coding from Juce.

I can never remember if comboBox::getSelectedId() ends up as comboBox.getSelectedId or comboBox:getSelectedId, and where we end up with brackets etc in LUA, though this is more due to my lack of knowledge, than a lack of documentation.

My search strategy is

Is there an example in ctrlr?
Is there an example in LUA? Yes - will it work in 5.1? Does it rely on a module like LuaJIT?
Is it documented in Juce?
